Zend/zend_closures.c

```diff
@@ -377,10 +377,49 @@ static ZEND_COLD zend_function *zend_closure_get_constructor(zend_object *object
 }
 /* }}} */
 
+static zend_always_inline int zend_closure_compare_fakes(zend_closure *lhs, zend_closure *rhs) { /* {{{ */
+	if (Z_TYPE(lhs->this_ptr) == IS_OBJECT && Z_TYPE(rhs->this_ptr) == IS_OBJECT) {
+		if (Z_OBJ(lhs->this_ptr) != Z_OBJ(rhs->this_ptr)) {
```

Comment:
@nikic should this use zend_compare_objects ?

Reply:
No. it should check identity here.

Reply:
Cool, just checking.